### PR TITLE
feat: add support for P curves and Ed curves

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Review by troyronda — CHANGES_REQUESTED** [review]
We are not switching to just P256.

File: cmd/orb-cli/common/common.go [hunk 1]
```diff
@@ -520,10 +520,18 @@ func (s *Signer) Headers() jws.Headers {
 		headers[jws.HeaderAlgorithm] = "EdDSA"
 	}
 
-	if s.publicKey.Crv == "P-256" || s.publicKey.Crv == "P-384" || s.publicKey.Crv == "P-521" {
+	if s.publicKey.Crv == "P-256" {
 		headers[jws.HeaderAlgorithm] = "ES256"
 	}
 
+	if s.publicKey.Crv == "P-384" {
```
Comment (sandrask) [inline cmd/orb-cli/common/common.go:527]: It would be nicer to have a map of crv to algorithm.

Comment (fqutishat) [reply cmd/orb-cli/common/common.go]: done

File: test/bdd/features/activitypub.feature [hunk 2]
```diff
@@ -314,35 +314,35 @@ Feature:
...
-    # Valid signature on GET with actor as a follower - should be allowed
-    When an HTTP GET is sent to "https://orb.domain1.com/services/orb/inbox" signed with KMS key from "domain2"
-    Then the JSON path "type" of the response equals "OrderedCollection"
-
-    # Valid signature on GET with actor as non-follower/non-witness - should be denied
-    When an HTTP GET is sent to "https://orb.domain1.com/services/orb/inbox" signed with KMS key from "domain3" and the returned status code is 401
-
-    And variable "undoFollowActivity" is assigned the JSON value '{"@context":"https://www.w3.org/ns/activitystreams","type":"Undo","actor":"${domain2IRI}","to":"${domain1IRI}","object":{"actor":"${domain2IRI}","id":"${followID}","object":"${domain1IRI}","type":"Follow"}}'
-    Then an HTTP POST is sent to "https://orb.domain2.com/services/orb/outbox" with content "${undoFollowActivity}" of type "application/json" signed with KMS key from "domain2"
+#  @activitypub_httpsig
```
[the remark this answers is not in the document]
Comment (fqutishat) [reply test/bdd/features/activitypub.feature]: remove by mistake

**Pull request merged**